src/other/index.jl

```diff
@@ -204,6 +204,9 @@ end
                         "DataFrame only supports indexing columns with integers, symbols or boolean vectors"))
 end
 
+@inline function Base.getindex(x::AbstractIndex, rx::Regex)
+    getindex(x, filter(name -> occursin(rx, String(name)), _names(x)))
+end
 # Helpers
```

Comment:
Add empty line.

Reply:
Added.

This PR is not ready for a review yet - I will let you know (and remove WIP) when its ready (I asked for your initial comments to make sure which way we go forward as there is a lot of tests to write). Thank you.